contracts/StreamManager.vy

```diff
@@ -252,3 +320,43 @@ def withdraw(creator: address, stream_id: uint256) -> uint256:
     log Withdrawn(creator, stream_id, funded_amount == withdrawal_amount, withdrawal_amount)
 
     return withdrawal_amount
+
+
+@external
+def batch_withdraw(
```

Comment:
Might be more expedient to pass in `token: ERC20` to this method, save the memory cost

Reply:
We still need to load the token on every iteration to asser the stream uses the same token
```
assert token.address == self.streams[creator][stream_id].token.address, "token must be the same for all streams"
```

Reply:
And the SLOAD is more expensive than writing one more address on-chain?

Reply:
Oh, no no, I mean add it as an input argument